pkg/controller/flow/resources/channel.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	channelsv1alpha1 "github.com/knative/eventing/pkg/apis/channels/v1alpha1"
+	"github.com/knative/eventing/pkg/apis/flows/v1alpha1"
+	"github.com/knative/eventing/pkg/controller"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func MakeChannel(defaultBusName string, flow *v1alpha1.Flow) *channelsv1alpha1.Channel {
+	channelName := flow.Name
+	channel := &channelsv1alpha1.Channel{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      channelName,
+			Namespace: flow.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*controller.NewControllerRef(flow),
```

Comment:
A possible alternative to this and other controller references is https://github.com/kubernetes-sigs/controller-runtime/blob/471f9cde2471949c85c01c3baba2947d9929c908/pkg/controller/controllerutil/controllerutil.go, which can generate a controller ref between any two metav1.Objects, no type case needed.

```go
import "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"

func MakeChannel(flow *v1alpha1.Flow) *channelsv1alpha1.Channel {
  channel := &channelsv1alpha1.Channel{}
  controllerutil.SetOwnerReference(flow, channel, scheme.Scheme)
  return channel
}
```

Reply:
Let's make a bug and a follow-up to fix this here and in serving. I can do this

Reply:
https://github.com/knative/eventing/issues/327